Orange/widgets/unsupervised/owmanifoldlearning.py

```diff
@@ -0,0 +1,284 @@
+    max_iter = Setting(300)
+    random_state = Setting(0)
+    random_state_values = (None, 0)
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.max_iter_spin = self._create_spin_parameter(
+            "max_iter", 10, 10 ** 4, "Max iterations:")
+        self.random_state_check = self._create_check_parameter(
+            "random_state", "Replicable (deterministic) learning")
+
+
+class IsomapParametersEditor(ManifoldParametersEditor):
+    n_neighbors = Setting(5)
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.n_neighbors_spin = self._create_spin_parameter(
+            "n_neighbors", 1, 10 ** 2, "Neighbors:")
+
+
+class LocallyLinearEmbeddingParametersEditor(ManifoldParametersEditor):
+    n_neighbors = Setting(5)
+    max_iter = Setting(100)
+    method_index = Setting(0)
+    method_values = (("standard", "Standard"),
+                     ("modified", "Modified"),
+                     ("hessian", "Hessian eigenmap"),
+                     ("ltsa", "Local"))
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.method_combo = self._create_combo_parameter(
+            "method", "Method:")
+        self.n_neighbors_spin = self._create_spin_parameter(
+            "n_neighbors", 1, 10 ** 2, "Neighbors:")
+        self.max_iter_spin = self._create_spin_parameter(
+            "max_iter", 10, 10 ** 4, "Max iterations:")
+
+
+class SpectralEmbeddingParametersEditor(ManifoldParametersEditor):
+    affinity_index = Setting(0)
+    affinity_values = (("nearest_neighbors", "Nearest neighbors"),
+                       ("rbf", "RBF kernel"))
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.affinity_combo = self._create_combo_parameter(
+            "affinity", "Affinity:")
+
+
+class OWManifoldLearning(OWWidget):
+    name = "Manifold Learning"
+    description = "Nonlinear dimensionality reduction."
+    icon = "icons/Manifold.svg"
+    priority = 2200
+
+    inputs = [("Data", Table, "set_data")]
+    outputs = [("Transformed data", Table)]
+
+    MANIFOLD_METHODS = (TSNE, MDS, Isomap, LocallyLinearEmbedding,
+                        SpectralEmbedding)
+    parameter_editors = (TSNEParametersEditor, MDSParametersEditor,
+                         IsomapParametersEditor,
+                         LocallyLinearEmbeddingParametersEditor,
+                         SpectralEmbeddingParametersEditor)
+
+    resizing_enabled = False
+    want_main_area = False
+
+    manifold_method_index = Setting(0)
+    n_components = Setting(2)
+    auto_apply = Setting(True)
+
+    class Error(OWWidget.Error):
+        n_neighbors_too_small = Msg("Neighbors must be greater than {}.")
+        negative_distances = Msg("All distances should be positive.")
+        manifold_error = Msg("{}")
+
+    def __init__(self):
+        self.data = None
+
+        # GUI
+        info_box = gui.vBox(self.controlArea, "Info")
```

Comment:
While the Info box at the top isn't inappropriate it is also not necessary IMHO.
And comparing this widget with similar ones: PCA, MDS, CA, k-means, hierarchical, or even any classification/regression widget - none of them have an Info box.